[WIP] Configure custom domains

tarteo opened the pull request:
![image](https://github.com/onesteinbv/addons-generic/assets/10028499/8c54254b-5fa5-4d09-bd98-5563a486bee3)
![image](https://github.com/onesteinbv/addons-generic/assets/10028499/5f57ef05-c252-45cd-9278-7c24c8600e9a)

tarteo committed:
argocd_deployer: Add generic key value pair (argocd.application.value)

tarteo committed:
[IMP] argocd_deployer: Add string to tag_ids

tarteo committed:
[REF] argocd_deployer: application urls
Use the yaml in argocd.application.config to get the urls of the application

tarteo committed:
argocd_deployer: Default domains for applications

tarteo committed:
argocd_website: user interface for configuring custom domain names

tarteo committed:
argocd_website: simple dns check

tarteo committed:
argocd_website: simple dns check

tarteo committed:
argocd_website: simple dns check improved ui

tarteo committed:
argocd_website: form validation

tarteo committed:
argocd_website: Fix small missing pre-commit error

tarteo committed:
Add account_asset_management to exclude list

tarteo committed:
arogcd_deployer: Add tests
argocd_sale: Add tests

ByteMeAsap commented on argocd_website/static/src/js/portal.js:
```diff
@@ -42,4 +42,74 @@ odoo.define("argocd_website.portal", function (require) {
+
+        _onKeyDown: function () {
+            this.$input.removeClass("is-invalid");
+            this.$input.removeClass("is-valid");
```
you can include both the classes at once separated by space removeClass("is-invalid is-valid")

ByteMeAsap commented on argocd_sale/tests/test_invoice_to_application.py:
```diff
@@ -0,0 +1,38 @@
+                "partner_id": partner.id,
+                "order_line": [
+                    Command.create(
+                        {"name": "Product", "product_id": product.id, "price_unit": 0}
```
Add [0] on product at product.id just to be sure and for readers

ByteMeAsap commented on argocd_website/static/src/js/portal.js:
```diff
@@ -42,4 +42,74 @@ odoo.define("argocd_website.portal", function (require) {
+
+            if (!domain) {
+                self.$input.removeClass("is-invalid");
+                self.$input.removeClass("is-valid");
```
you can include both the classes at once separated by space removeClass("is-invalid is-valid")

tarteo commented:
TODO: Forgot to check access rights for portal users

tarteo committed:
Smaller load icon

tarteo committed:
Small UX improvement and access rules for portal users

tarteo committed:
Merge branch '16.0' into 16.0-custom-domain

tarteo commented:
I want to add A record support

MrGigSolutions committed:
[IMP] argocd_deployer: application_set is now a model

Added the application_set model to support multiple application sets.
Also cleaned up tha application code a bit.

MrGigSolutions committed:
[IMP] argocd_deployer: Added new domain computation to application set.

MrGigSolutions committed:
[IMP] argocd_sale: Support for application sets

MrGigSolutions committed:
Merge remote-tracking branch 'origin/16.0' into 16.0-custom-domain

# Conflicts:
#	argocd_sale/demo/product_template_demo.xml
#	argocd_sale/models/account_move.py
#	argocd_sale/tests/__init__.py
#	argocd_website/README.rst

MrGigSolutions committed:
[IMP] argocd_sale: Support for application sets

MrGigSolutions committed:
[IMP] argocd_deployment: Support for "application set application sets".

MrGigSolutions committed:
[FIX] argocd_deployment: Applications now deploy in the correct folder.

MrGigSolutions committed:
[FIX] argocd_deployment - Fixed warnings and tests.

MrGigSolutions committed:
[UPD] argocd_deployment - Improved README.rst

MrGigSolutions committed:
[FIX] argocd_deployment - Added templates to application sets.

MrGigSolutions committed:
[FIX] argocd_deployment - Fix to app set form.

MrGigSolutions committed:
[FIX] argocd_deployer - Fix for is_master_deployment.

MrGigSolutions committed:
[FIX] argocd_deployer - Only one master set allowed and fixed push

MrGigSolutions committed:
[IMP] argocd_deployer - Add a Chart.yaml file to the master application set

MrGigSolutions committed:
[FIX] argocd_deployer - Fix master template to be usable with Helm/ArgoCD

MrGigSolutions committed:
[FIX] argocd_deployer - Fix master template to be usable with Helm/ArgoCD and fixed commit message.

tarteo committed:
Temporarily disable configuring domain names

MrGigSolutions committed:
[IMP] argocd_deployer - Improvements to usage of application sets

- Renamed is_master_deployment to is_master
- added domain YAML overrides to application tags to be more flexible with changes in YAML layout in different application sets
- application now has an is_deployed field that's True when both it and its application set are deployed
- application now has an is_application_set_deployed field to indicate whether its application set is deployed
- added nice colors to the lists of applications and application sets to indicate they are deployed
- Added various jumbotron warnings to applications and application sets
- Added a setting to disable direct deployment
- Added a delay to application set destruction when calling through the job queue
- It's now only possible to destroy application sets with deployed applications through the job queue

MrGigSolutions committed:
[IMP] argocd_deployer - Customizable delay for destroying applications and application sets.

MrGigSolutions committed:
Merge branch '16.0' into 16.0-custom-domain

MrGigSolutions committed:
[FIX] argocd_deployer - Typo using wrong config value.

MrGigSolutions reviewed (approved):
Approved for testing.
